application/app/controllers/files_controller.rb

```diff
@@ -29,6 +30,35 @@ def cancel
     head :no_content
   end
 
+  def cancel_upload
```

Comment:
we need to move this method as part of the refactoring. Will decide how to structure the upload and download controllers